File: day11/.ts
function findUnsafeGifts(warehouse: string[]): number {
  const warehouseBoard = warehouse.filter((line) => line.trim() != "");
  const rows = warehouseBoard.length;
  const cols = warehouseBoard[0].length;
  let unprotectedGifts = 0;
  for (let row = 0; row < rows; row++) {
    for (let col = 0; col < cols; col++) {
      if (warehouseBoard[row][col] == "*") {
        const up = row > 0 && warehouseBoard[row - 1][col] == "#";
        const down = row < rows - 1 && warehouseBoard[row + 1][col] == "#";
        const left = col > 0 && warehouseBoard[row][col - 1] == "#";
        const right = col < cols - 1 && warehouseBoard[row][col + 1] == "#";
        if (!(up || down || left || right)) {
          unprotectedGifts++;
        }
      }
    }
  }
  return unprotectedGifts;
}

File: day11/.py
def find_unsafe_gifts(warehouse: list[str]) -> int:
  warehouse_board = [line for line in warehouse if line.strip()]
  rows = len(warehouse_board)
  cols = len(warehouse_board[0])
  unprodected_gifts = 0

  for row in range(rows):
    for col in range(cols):
      if warehouse_board[row][col] == "*":
        up = row > 0 and warehouse_board[row-1][col] == "#"
        down = row < rows - 1 and warehouse_board[row+1][col] == "#"
        left = col > 0 and warehouse_board[row][col - 1] == "#";
        right = col < cols - 1 and warehouse_board[row][col + 1] == "#";

        if not (up or down or left or right):
          unprodected_gifts += 1
  return unprodected_gifts
